### ISSUE TITLE
`Type[Enum]` does not properly infer from `classmethod` when passed to `list`

### ISSUE THREAD

**Issue by johnthagen** [python/mypy#14110]
**Bug Report**

`Type[Enum]` does not properly infer when passed to `list`,  and is inferred to `Any` or `str` in the case of a `(str, Enum)` type.

**To Reproduce**

```python
from enum import Enum, unique

@unique
class Color(Enum):
    Red = "red"
    Green = "green"
    Blue = "blue"

    @classmethod
    def foo(cls) -> None:
        colors = list(cls)
        reveal_type(colors)
        print(colors)


Color.foo()
```

**Expected Behavior**

Revealed type:  `builtins.list[main.Color]`

Running the code, shows that at runtime it is a `list[Color]`:

```
[<Color.Red: 'red'>, <Color.Green: 'green'>, <Color.Blue: 'blue'>]
```

**Actual Behavior**

```
main.py:12: error: Need type annotation for "colors" (hint: "colors: List[<type>] = ...")  [var-annotated]
main.py:13: note: Revealed type is "builtins.list[Any]"
Found 1 error in 1 file (checked 1 source file)
```

**Your Environment**

- Mypy version used: `mypy 0.991 (compiled: yes)`
- Mypy command-line flags: None
- Mypy configuration options from `mypy.ini` (and other config files): N/A (default settings)
- Python version used: 3.10.6
- OS: macOS 12.6, x86_64

**Comment by johnthagen** [comment]
When a `(str, Enum)` type is used:

```py
from enum import Enum, unique


@unique
class Color(str, Enum):
    Red = "red"
    Green = "green"
    Blue = "blue"

    @classmethod
    def foo(cls) -> None:
        colors = list(cls)
        reveal_type(colors)
        print(colors)


Color.foo()
```

The revealed type is `list[str]` instead of `list[Color]`:

```
main.py:13: note: Revealed type is "builtins.list[builtins.str]"
Success: no issues found in 1 source file
```

Even though at runtime, it is a `list[Color]`:

```
[<Color.Red: 'red'>, <Color.Green: 'green'>, <Color.Blue: 'blue'>]
```

**Comment by erictraut** [comment]
This bug appears to have been fixed in mypy 1.1.0. I'm not able to repro it in that version or any subsequent version.

**Comment by hauntsaninja** [comment]
Fixed in #14554
